master/internal/authz/obfuscate.go

```diff
@@ -120,3 +123,60 @@ func ObfuscateJob(job *jobv1.Job) jobv1.LimitedJob {
 		WorkspaceId:    job.WorkspaceId,
 	}
 }
+
+// ObfuscateExperiments obfuscates sensitive information in experiments.
+// Currently, that is considered to be anything the user has configured
+// under a "secrets" key in the general-purpose "data" config.
+func ObfuscateExperiments(experiments ...*experimentv1.Experiment) {
+	for _, exp := range experiments {
+		data, exists := exp.Config.Fields["data"] //nolint:staticcheck
+		if !exists {
+			continue
+		}
+		dataMap := data.GetStructValue() // nil if not struct
+		if dataMap == nil {
+			continue
+		}
+		secrets, exists := dataMap.Fields["secrets"]
+		if !exists {
+			continue
+		}
+		secretsMap := secrets.GetStructValue()
+		if secretsMap == nil {
+			continue
+		}
+		for key := range secretsMap.Fields {
+			secretsMap.Fields[key] = structpb.NewStringValue(hiddenString)
+		}
+
+		var oConfig map[string]interface{}
+		err := json.Unmarshal([]byte(exp.OriginalConfig), &oConfig)
```

Comment:
Probably nearly certainly always a non-issue, but since leaking secrets is potentially a Big Deal, is it worth having this return an error on failure so calling code can return an error rather than returning unobfuscated experiments? We'd probably have to hit some kind of strange OOM error in `json.Unmarshal` or something for this to return an error when we pass in valid JSON, so maybe it's not worth it. What do you think?

Reply:
We discussed in Slack and concluded that yes, we should return an error here.